cli/QuickstepCli.cpp

```diff
@@ -135,9 +141,22 @@ int main(int argc, char* argv[]) {
   gflags::ParseCommandLineFlags(&argc, &argv, true);
   google::InitGoogleLogging(argv[0]);
 
-  if (quickstep::FLAGS_num_workers > 0) {
-    printf("Starting Quickstep with %d worker thread(s)\n",
-           quickstep::FLAGS_num_workers);
+  // Detect the hardware concurrency level. Note this call will return 0
+  // if it fails (which it may on some machines/environments).
+  const unsigned int num_hw_threads = std::thread::hardware_concurrency();
+
+  // Use the command-line value if that was supplied, else use the value
+  // that we computed above, provided it did return a valid value.
+  // TODO(jmp): May need to change this at some point to keep one thread
+  //            available for the OS if the hardware concurrency level is high.
```

Comment:
Do we need to reserve one hardware thread for `Foreman`? And one for `Shiftboss` in near future?

FYI, `Foreman` will be changed as a long-running service to process concurrent query executions, instead of current one-query-per-Foreman model.